### PR TITLE
feat(CNAD): add a new resource cnad advanced policy ip list

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:
add a new resource cnad advanced policy black and white list
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

- This PR depends on #3158

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cnad' TESTARGS='-run TestAccBlackWhiteList_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/cnad -v -run TestAccBlackWhiteList_basic -timeout 360m -parallel 4 
=== RUN   TestAccBlackWhiteList_basic 
=== PAUSE TestAccBlackWhiteList_basic
=== CONT  TestAccBlackWhiteList_basic
--- PASS: TestAccBlackWhiteList_basic (26.74s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cnad      26.809s
```

### CODE REVIEW

File: huaweicloud/services/cnad/resource_huaweicloud_cnad_advanced_black_white_list.go [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+func resourceBlackWhiteListUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	client, err := cfg.NewServiceClient("aad", "")
+	if err != nil {
+		return diag.Errorf("error creating CNAD Client: %s", err)
+	}
+
+	if d.HasChanges("black_ip_list") {
+		oldRaws, newRaws := d.GetChange("black_ip_list")
```
Comment (houpeng80) [inline huaweicloud/services/cnad/resource_huaweicloud_cnad_advanced_black_white_list.go:135]: how about change the type to schema.Set, the addIpList and deleteIpList can be get by the function of Difference and the unchanged ips will not be delete and add again

File: docs/resources/cnad_advanced_black_white_list.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `policy_id` - (Required, String, ForceNew) Specifies the policy id.
+
+  Changing this parameter will create a new resource.
+
+* `black_ip_list` - (Optional, List) Specifies the black ip list. The minimum length is 7. The maximum length is 128.
```
Comment (hmbig2) [inline docs/resources/cnad_advanced_black_white_list.md]: What's the meaning: The minimum length is 7. There is only two ip in the acc test.

Comment (deer-hang) [reply docs/resources/cnad_advanced_black_white_list.md]: I have changed the description, The length indicate the IP address or IP address range.

File: huaweicloud/services/cnad/resource_huaweicloud_cnad_advanced_black_white_list.go (hunk 1 above)
Comment (deer-hang) [reply huaweicloud/services/cnad/resource_huaweicloud_cnad_advanced_black_white_list.go:135]: OK, I have change it to `TypeSet`

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cnad' TESTARGS='-run TestAccBlackWhiteList_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/cnad -v -run TestAccBlackWhiteList_basic -timeout 360m -parallel 4 
=== RUN   TestAccBlackWhiteList_basic 
=== PAUSE TestAccBlackWhiteList_basic
=== CONT  TestAccBlackWhiteList_basic
--- PASS: TestAccBlackWhiteList_basic (26.54s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cnad      26.602s
```

**Comment by deer-hang** [comment]
```
make testacc TEST='./huaweicloud/services/acceptance/cnad' TESTARGS='-run TestAccBlackWhiteList_basic'
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/cnad -v -run TestAccBlackWhiteList_basic -timeout 360m -parallel 4 
=== RUN   TestAccBlackWhiteList_basic 
=== PAUSE TestAccBlackWhiteList_basic
=== CONT  TestAccBlackWhiteList_basic
--- PASS: TestAccBlackWhiteList_basic (23.66s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cnad      23.726s
```

**Pull request merged**